README.md

```diff
@@ -340,7 +340,7 @@ Let’s create two new functions that let us deposit and withdraw liquidity. How
 
 ### 🥅 Goals / Checks
 
-- [ ] 🥅 Before using the deposit function, check the required amount of $BAL tokens, which is equal to the amount of ETH you have entered for deposit by using the price function in debug tab.
+- [ ] 🥅 Before using the deposit function, check the required amount of $BAL tokens, which is equal to the amount of ETH you have entered for deposit by using the price function in debug tab. Now, you can use `approve()` function with only required amount of tokens. 
```

Comment:
For me it's difficult to understand what do you mean.

As I understand, required amount is equal to `tokenDeposit = (msg.value * tokenReserve / ethReserve) + 1;`, isn't it? See `deposit()` function.
It's close to result of `price()`, but without fee

Reply:
What about adding a side quest in the goals, something like this:

### 🥅 Goals / Checks

- [ ] 💧 Deposit liquidity, and then check your liquidity amount through the mapping in the debug tab. Has it changed properly? Did the right amount of assets get deposited?
> ⚔️ Side Quest:  Prove your knowledge by calculating the exact amount of Balloons you need to approve when depositing. Don't forget about the fee!
- [ ] 🧐 What happens if you `deposit()` at the beginning of the deployed contract, then another user starts swapping out for most of the balloons, and then you try to withdraw your position as a liquidity provider? Answer: you should get the amount of liquidity proportional to the ratio of assets within the isolated liquidity pool. It will not be 1:1.

Reply:
> What about adding a side quest in the goals, something like this:
> Deposit liquidity, and then

Good Idea, but again amount that needs to be approved before a deposit is still unclear if ratio is not 1:1.

Reply:
> For me it's difficult to understand what do you mean.
> 
> As I understand, required amount is equal to `tokenDeposit = (msg.value * tokenReserve / ethReserve) + 1;`, isn't it? See `deposit()` function. It's close to result of `price()`, but without fee

hmm I thought that the price exactly work as we use in ethToToken or tokenToEth functions, you are right. I think showing the token amount in deposit function could be really helpful for new users to make them understand the logic better. The point is that if we want them to use approve function with exactly correct amount of BAL tokens, we might need to consider about deleting the approving the high amount of tokens at the beginning of the challenge. I think the thing you said about approve amount yesterday was right ser @ZakGriffith . What do you think

Reply:
not sure how they need to calculate the exact amount, manually? and how to check if they're right? by allowance amount left after deposit?

Reply:
Yeah it can be done by getting the difference between the allowance before using deposit and after using the deposit function. Maybe by creating a pop-up that includes "... AMOUNTS OF $BAL WERE USED"

Or we can create a new function in smart contract for getting the token amount in deposit function just like the price function but specified for deposit.

Maybe we can find a better solution :)

Reply:
Here is a question, is there any real reason why someone would need to only approve the exact amount instead of an arbitrary larger number?  Maybe we're over thinking this.

Reply:
just to understand how it works 🤷 . Probably it's redundant, yes